heat/core/arithmetics.py

```diff
@@ -776,6 +776,26 @@ def pow(t1: Union[DNDarray, float], t2: Union[DNDarray, float]) -> DNDarray:
     DNDarray([[ 1.,  8.],
             [27., 64.]], dtype=ht.float32, device=cpu:0, split=None)
     """
+    if isinstance(t1, DNDarray):
+        ret = factories.zeros_like(t1)
+        try:
+            t2 = manipulations.resplit(t2, t1.split)
```

Comment:
Hey I'm thinking this might backfire. Example: two DNDarrays `t1` and `t2`. 
`t1.shape = (100, 8)` and `t1.split=1`. 
`t2.shape = (8,)`.

The shapes are broadcastable and `ht.pow(t1, t2)` is legitimate. `t2` cannot be split along 1 though, so in the follow-up the assumption is that `t2` is not a DNDarray. Line 786 will fail. 

I'm wondering if we need so many checks at all. All we need to know is if `t1` or `t2` is a scalar (with the other one a DNDarray). How about 
```python
# t1 is a verified DNDarray
if not hasattr(t2, "dtype"):
   ret = torch.pow(t1.larray, t2)
else:
   # call binary_op
```



 or something like that? What am I missing?

Reply:
you are correct. that was an issue with the assumption that i made. i updated the logic in this to correct for this now. it should be able to handle anything you can throw at it (i hope).